tests/system/python/packages/test_authentication.py

```diff
@@ -0,0 +1,1157 @@
+        r = r.read().decode()
+        jdoc = json.loads(r)
+        assert {'message': 'User with id:<5> has been updated successfully'} == jdoc
+
+    def test_delete_user_with_password_token(self, foglamp_url):
+        conn = http.client.HTTPConnection(foglamp_url)
+        conn.request("DELETE", "/foglamp/admin/4/delete", headers={"authorization": PASSWORD_TOKEN})
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        jdoc = json.loads(r)
+        assert {'message': "User has been deleted successfully"} == jdoc
+
+    def test_delete_user_with_certificate_token(self, foglamp_url):
+        conn = http.client.HTTPConnection(foglamp_url)
+        conn.request("DELETE", "/foglamp/admin/6/delete", headers={"authorization": CERT_TOKEN})
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        jdoc = json.loads(r)
+        assert {'message': "User has been deleted successfully"} == jdoc
+
+    def test_logout_all_with_password_token(self, foglamp_url):
+        conn = http.client.HTTPConnection(foglamp_url)
+        conn.request("PUT", '/foglamp/1/logout', headers={"authorization": PASSWORD_TOKEN})
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        jdoc = json.loads(r)
+        assert jdoc['logout']
+
+    def test_admin_actions_reg_user_with_password_token(self, foglamp_url):
+        """Test that regular user is not able to perform any actions that only an admin can"""
+        # Login with regular user
+        conn = http.client.HTTPConnection(foglamp_url)
+        conn.request("POST", "/foglamp/login", json.dumps({"username": "user", "password": "foglamp"}))
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        jdoc = json.loads(r)
+        assert not jdoc['admin']
+        _token = jdoc["token"]
+
+        # Create User
+        conn.request("POST", "/foglamp/admin/user", body=json.dumps({"username": "other",
+                                                                     "password": "User@123"}),
+                     headers={"authorization": _token})
+        r = conn.getresponse()
+        assert 403 == r.status
+        r = r.read().decode()
+        assert "403: Forbidden" == r
+
+        # Reset User
+        conn.request("PUT", "/foglamp/admin/2/reset", body=json.dumps({"role_id": 1, "password": "F0gl@p!"}),
+                     headers={"authorization": _token})
+        r = conn.getresponse()
+        assert 403 == r.status
+        r = r.read().decode()
+        assert "403: Forbidden" == r
+
+        # Delete User
+        conn.request("DELETE", "/foglamp/admin/2/delete", headers={"authorization": _token})
+        r = conn.getresponse()
+        assert 403 == r.status
+        r = r.read().decode()
+        assert "403: Forbidden" == r
+
+    def test_admin_actions_reg_user_with_certificate_token(self, foglamp_url):
```

Comment:
test_admin_actions_reg_user_with_regular_user_certificate (And, No _token with certificate)